Mathlib/Geometry/Manifold/MFDeriv/SpecificFunctions.lean

```diff
@@ -377,7 +377,8 @@ theorem MDifferentiableAt.mfderiv_prod {f : M → M'} {g : M → M''} {x : M}
       (mfderiv I I' f x).prod (mfderiv I I'' g x) := by
   classical
   simp_rw [mfderiv, if_pos (hf.prod_mk hg), if_pos hf, if_pos hg]
-  exact hf.2.fderivWithin_prod hg.2 (I.unique_diff _ (mem_range_self _))
+  exact hf.differentiableWithinAt_writtenInExtChartAt.fderivWithin_prod
```

Comment:
Expanding `.1` and `.2`  into e.g. `differentiableWithinAt_writtenInExtChartAt` makes the code *much* more verbose, but also more explicit. I'm not sure which one I prefer, but I don't have a strong opinion either way.

Reply:
I wouldn't mind picking a shorter lemma name, but I don't have a particular idea either.

Reply:
see PR message. I also wouldn't mind a shorter name.